sdk/rust/src/http.rs

```diff
@@ -15,13 +15,132 @@ pub use super::wit::wasi::http::types::*;
 /// is no need for streaming bodies.
 pub struct Request {
     /// The method of the request
-    pub method: Method,
-    /// The path together with the query string
-    pub path_and_query: String,
+    method: Method,
+    /// The uri for the request
+    ///
+    /// The first item is set to `None` if the supplied uri is malformed
+    uri: (Option<hyperium::Uri>, String),
```

Comment:
if we haven't use hyper's `Uri` type before, then I think we should strongly consider using [rust-url](https://crates.io/crates/url) instead. It's about as battle-tested as can be. Importantly it's what Spin uses on the host-side as well, meaning there's no risk of security issues caused by varying interpretations of URLs.

Reply:
Ok, I guess we can live with this for now. The security issues I'm talking about are real, unfortunately, but are much more likely to affect very naive, untested implementations, neither of which Hyper's is.